simple-factsheet.Rmd

```diff
@@ -23,91 +28,401 @@ library(tidyverse)
 library(knitr)
 library(kableExtra)
 library(urbnthemes)
+library(here)
+library(scales)
+library(readxl)
+library(stringr)
+library(ggrepel)
 
 set_urbn_defaults(style = "print")
 ```
 
+```{r plots, include=FALSE, results="hide"}
+filepath <- here("/data/factsheet_who are-support_data_states_201718_varnames.xlsx")
+
+topercent <- function(x) {
+  if(abs(x) > 0.01) {
+    round((x * 100),0)
+  } else{
+    round((x * 100),1)
+  }
+} 
+
+state_text <- ifelse(params$state == "District of Columbia",
+                     paste("the",params$state,sep=" "),
+                     params$state)
+
+states_subset <- read_excel(filepath) %>%
+  filter(StateName == params$state) %>%
+  rename_at(vars(contains("1718")),~sub("_1718", "", .)) %>%
+  
+  # prepare overall cOI population number - rounded thousand or numeric + "million" text
+  mutate(text_1_popcount = case_when(total_popcount_coi < 1000000 ~ formatC(total_popcount_coi,
+                                                                            format ="f", 
+                                                                            digits = 0,
+                                                                            big.mark =","),
+                                     TRUE ~ paste(round((total_popcount_coi/1000000),1), "million"))) %>%
+  
+  # convert decimals to be communicated as percentages - no decimals except under 1% 
+  # Except for those that will only be used in data vis 
+  mutate_at(vars(starts_with("pct"),
+                 starts_with("share")), topercent) %>%
+  
+  # prepare strings that'll change
+  mutate(text_2_citizens = case_when(share_uscitizen_coi >= 85 ~ "Almost all",
+                                     share_uscitizen_coi < 85 & share_uscitizen_coi >= 50 ~ "Most",
+                                     share_uscitizen_coi < 50 ~ "Many",
+                                     TRUE ~ NA_character_),
+         
+         text_3_citizens = case_when(share_prtnatcit_coi >= 50 ~ "most",
+                                     share_prtnatcit_coi < 50 ~ "many",
+                                     TRUE ~ NA_character_),
+         
+         temp_working_diff = share_workingfam_coi - share_workingfam_nat,
+         
+         text_4_employment = case_when(abs(temp_working_diff) < 3 ~ "similar for children of immigrants and",
```

Comment:
`case_when()` evaluates the logical conditions in order from top to bottom. The following is a clearer way to write this section:

```
case_when(temp_working_diff <= -6     ~ "lower for children of immigrants than",
                  temp_working_diff <= -3     ~ "slightly lower for children of immigrants than",
                  temp_working_diff < 3 ~ "similar for children of immigrants and",
                  temp_working_diff < 6         ~ "slightly higher for children of immigrants than",
                  TRUE                                     ~ "higher for children of immigrants than")
```

Reply:
got it; thank you.